axiom/store.py

```diff
@@ -2309,7 +2303,7 @@ def getItemByID(self, storeID, default=_noItem, autoUpgrade=True):
                 self.objectCache.cache(storeID, x)
             return x
         if default is _noItem:
-            raise KeyError(storeID)
+            raise errors.ItemNotFound(storeID)
```

Comment:
Should probably update the method's docstring to note that this raises ItemNotFound, not KeyError.

Reply:
Are there any consequences emanating from this change?  I think ItemNotFound derives from KeyError, so probably nothing breaks, but it might be good to ensure that code that calls or tests getItemByID play nicely with ItemNotFound.

Reply:
Yes, `ItemNotFound` is a subclass of `KeyError`, so it's quite hard to write code that accidentally relies on the specific exception type (and in fact the other branch already raises `ItemNotFound`). I've run my day job's test suite successfully against this branch, as well as the Mantissa and Imaginary test suites, so I'm reasonably confident in this change. Good point about the docstring though.

Reply:
Ok, sounds good to me.